doc/source/whatsnew/v2.3.0.rst

```diff
@@ -108,6 +108,7 @@ Performance improvements
 
 Bug fixes
 ~~~~~~~~~
+Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` for no arg validation unlike :meth:`Dataframe.diff` for periods. (:issue:`56607`)
```

Comment:
```suggestion
Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)
```

Reply:
Shouldn't it be allowing only integer values @lithomas1 ?

Reply:
Done @lithomas1